lib/dal/src/job/definition/dependent_values_update.rs

```diff
@@ -188,6 +194,68 @@ impl DependentValuesUpdate {
 
         let mut update_tasks = JoinSet::new();
 
+        // This is the core loop. Use both the individual and the management subscription to determine what to do next.
+        let needs_restart = tokio::select! {
+            result = self.listen(ctx, council, pub_council, dependency_graph, &mut status_updater, &mut update_tasks) => result,
+            management_result = self.listen_management(&mut management_council) => management_result,
+        }?;
+
+        // If we need to restart, we need to stop what we are doing, tell the status update that we are stopping, and
+        // then re-enqueue ourselves.
+        if needs_restart {
+            update_tasks.abort_all();
+
+            info!(?self.attribute_values, "aborted update tasks and restarting job");
+
+            status_updater
+                .values_completed(ctx, self.attribute_values.clone())
+                .await;
+
+            ctx.enqueue_job(DependentValuesUpdate::new(
+                self.access_builder,
+                self.visibility,
+                self.attribute_values.clone(),
+            ))
+            .await?;
+        }
```

Comment:
Here's what a restart entails:
- Abort all tasks
- Log what's happening
- Mark all as completed with the status updated
- Re-enqueue the job